ndscan/experiment/default_analysis.py

```diff
@@ -197,18 +247,19 @@ def __init__(
         for channel in analysis_results:
             name = channel.path
             if name in self._result_channels:
-                axes = ", ".join(h._store.identity for h in self._required_axis_handles)
-                raise ValueError("Duplicate analysis result channel name '" + name +
-                                 "' in analysis for axes '" + axes + "'")
+                axes = ", ".join(h.name + "@" + h.owner._stringize_path()
+                                 for h in self._required_axis_handles)
+                raise ValueError(f"Duplicate analysis result channel name '{name}' " +
+                                 f"in analysis for axes [{axes}]")
```

Comment:
```suggestion
                raise ValueError(f"Duplicate analysis result channel name '{name}' "
                                 f"in analysis for axes [{axes}]")
```

No need for the `+` here.

Reply:
Please keep to the existing coding style. Implicit string concatenation is evil; I'm still hoping the Python core devs will at some point see the light and remove it (e.g. as [suggested](https://lwn.net/Articles/551438/) by Guido himself as well).